sklearn/cluster/birch.py

```diff
@@ -67,8 +67,9 @@ def _split_node(node, threshold, branching_factor):
         if node.next_leaf_ is not None:
             node.next_leaf_.prev_leaf_ = new_node2
 
+    squared_norm = node.squared_norm_[np.newaxis, :]
     dist = euclidean_distances(
-        node.centroids_, Y_norm_squared=node.squared_norm_, squared=True)
+        node.centroids_, Y_norm_squared=squared_norm, squared=True)
```

Comment:
Many things were internally converted to 2D using the `np.atleast2d` call, without it, many tests are causing errors.